### PR TITLE
Feature/45 add field stars #45

### PR DESCRIPTION
**Pull request by tykayn**
Adds an array of stars in the POI card, and a Field component presenting stars.

Screenshot 2023-01-27 at 14-05-54 Carte Touristique des Landes - Carte Touristique des Landes

### CODE REVIEW

File: components/PoisCard/PoiCard.vue [hunk 1]
```diff
@@ -203,6 +210,7 @@ export default Vue.extend({
     }),
 
     id(): ApiPoiId {
+      console.log('this.poi.properties', this.poi.properties)
```
Comment (frodrigo) [inline components/PoisCard/PoiCard.vue:213]:
```suggestion
```

File: components/PoisCard/PoiCard.vue [hunk 2]
```diff
@@ -241,6 +248,9 @@ export default Vue.extend({
     description(): string | undefined {
       return this.poi.properties.description
     },
+    stars(): number | undefined {
```
Comment (frodrigo) [inline components/PoisCard/PoiCard.vue:251]:
```suggestion

    stars(): number | undefined {
```

File: components/PoisCard/PoiCard.vue [hunk 3]
```diff
@@ -64,7 +64,14 @@
 
         {{ category }}
       </div>
+      <div :title="`${stars} ${$tc('fields.stars.label')}`" class="stars-data text-amber-500" v-if="stars">
```
Comment (frodrigo) [inline components/PoisCard/PoiCard.vue:68]:
```suggestion
```

File: components/PoisCard/PoiCard.vue [hunk 4]
```diff
@@ -64,7 +64,14 @@
...
 
+        <span :key="index" class="star-item" v-for="index in new Array(stars*1)">
+          <font-awesome-icon
+            :icon="'star'"
```
Comment (frodrigo) [inline components/PoisCard/PoiCard.vue:71]: But that does not use `Fields/Stars.vue` ?

File: components/Fields/Stars.vue [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  Vue.extend({
+
+    props: {
+      stars: {
+        type: Number,
+        default: 0,
+      },
+    },
+  })
+
```
Comment (frodrigo) [inline components/Fields/Stars.vue:28]: Should start at the start of the line, no identation.

File: components/PoisCard/PoiCard.vue (hunk 4 above)
Comment (tykayn) [reply components/PoisCard/PoiCard.vue:71]: we can do this, i saw other fields of the poi card that were not using other components and thought it was what we do in this place.
will fix that

File: components/Fields/Stars.vue (hunk 5 above)
Comment (frodrigo) [reply components/Fields/Stars.vue:28]: Better to have an indeps component to test it.

File: components/PoisCard/PoiCard.vue (hunk 4 above)
Comment (frodrigo) [reply components/PoisCard/PoiCard.vue:71]: Better to have an indeps component to test it.

**Comment by tykayn** [comment]
i tested with mock values that the stars are showing properly, is there some POI with this data in the set of the bbox of the project ?

File: cypress/fixtures/teritorio/references/pois/category/211.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
+{
```
Comment (frodrigo) [inline cypress/fixtures/teritorio/references/pois/category/211.json:1]: Do not commit theses files

**Comment by frodrigo** [comment]
Please add corresponding storiesbook.

File: components/Fields/Stars.stories.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  * to learn how to generate automatic titles
+  */
+  title: 'Stars',
+  component: Stars,
+};
+
+export const Primary = () => ({
+  components: { Stars },
+  template: '<Stars stars="3" />',
+});
```
Comment (frodrigo) [inline components/Fields/Stars.stories.ts:16]: Not required in other.
Really need here?

File: components/Fields/Stars.stories.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+export const Default = bind(Stars, {
+  ...defaultProps,
+})
+
+export const More = bind(Stars, {
+  stars:10,
+})
+export const None = bind(Stars, {
+  stars:0,
```
Comment (frodrigo) [inline components/Fields/Stars.stories.ts:30]: Missing space should not pass the code lint.

**Comment by frodrigo** [comment]
Please, can you rebase and clean the history of this branch, remove the commits of the fixtures file too.

File: components/Fields/Stars.stories.ts (hunk 8 above)
Comment (tykayn) [reply components/Fields/Stars.stories.ts]: linted and amended

**Comment by tykayn** [comment]
i do not get why the storybook step fails on the CI,
when i run "yarn build:storybook" locally it works.

File: components/Fields/Stars.stories.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+   */
+  title: 'Stars',
+  component: Stars,
+}
+
+export const Primary = () => ({
+  components: { Stars },
+  template: '<Stars stars="3" />',
+})
+
```
Comment (frodrigo) [inline components/Fields/Stars.stories.ts:18]: Stil to be remodev.

File: components/Home/Home.vue [hunk 10]
```diff
@@ -1,5 +1,24 @@
 <template>
   <div class="fixed w-full h-full overflow-hidden flex flex-col">
+    <t-full-modal ref="detailModal" :hide-close-button="true">
```
Comment (frodrigo) [inline components/Home/Home.vue:3]: This change has still nothing to to here.

Comment (tykayn) [reply components/Home/Home.vue:3]: this modal comes from the branch dev2

Comment (frodrigo) [reply components/Home/Home.vue]: I just switched the PR to target dev2.

File: components/Fields/Stars.stories.ts (hunk 9 above)
Comment (tykayn) [reply components/Fields/Stars.stories.ts:18]: removed

**Comment by tykayn** [comment]
are the stars info fetched in Vido?
we can see there are a lot of these in overpass 
https://overpass-turbo.eu/s/1qOn

**Comment by tykayn** [comment]
> > i do not get why the storybook step fails on the CI, when i run "yarn build:storybook" locally it works.
> 
> I guess it is because there is new files, PNG have to be added to git.
> 
> ```
> ✚ append  __screenshots__/current/Fields/Field/Stars.png
> ✚ append  __screenshots__/current/Stars/Default.png
> ✚ append  __screenshots__/current/Stars/More.png
> ✚ append  __screenshots__/current/Stars/None.png
> ✚ append  __screenshots__/current/Stars/Primary.png
> ```

storybook screenshots are ignored in gitignore:
```
# Storybook
.nuxt-storybook
storybook-static
__screenshots__/current
__screenshots__/diff
``̀`
should i remove this instruction to keep all current images ?

File: components/Fields/Stars.stories.ts (hunk 7 above)
Comment (tykayn) [reply components/Fields/Stars.stories.ts]: removed

File: components/PoisCard/PoiCard.vue [hunk 11]
```diff
@@ -65,7 +65,7 @@
 
         {{ category }}
       </div>
-
+      <Stars :stars="stars"></Stars>
```
Comment (frodrigo) [inline components/PoisCard/PoiCard.vue:68]: Should go here https://github.com/teritorio/vido/blob/feature/clear-all-categories/components/Fields/Field.vue

File: components/Fields/Field.vue [hunk 12]
```diff
@@ -21,6 +21,7 @@
       :class="`field_header_level_${recursionStack.length}`"
       >{{ fieldTranslateK(field.field) }}</FieldsHeader
     >
+    <Stars v-if="stars" :stars="stars"></Stars>
   </AddressField>
```
Comment (frodrigo) [inline components/Fields/Field.vue:25]: Not into AddressField!

```suggestion
  </AddressField>

  <Stars v-if="stars" :stars="stars"></Stars>
```

Comment (tykayn) [reply components/Fields/Field.vue:25]: i tried to add it outside of the other fields, but Vue complains about having only one root element.
i can wrap all fields in a div to have only one root element, other fields seems to be mutually exclusive

Comment (frodrigo) [reply components/Fields/Field.vue:25]: This one is also mutually exclusive : `v-else-if="field.field == 'starts'"`

**Comment by frodrigo** [comment]
Target is now `develop` branch.

**Comment by frodrigo** [comment]
Manually cleaned and merged 8dd7041d9a3f18d5657331568e16eef23a3abfa1

**Pull request closed without merging**